### ISSUE TITLE
sql: COPY does not honor the timezone

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#46250]
tldr: The input data for COPY does not honor timezones when used to populate tz/date/time columns.

This is a buglet due to the weird way the `planner` object gets initialized for COPY. I noticed this while working on #46249 

- in `(c *copyMachine) addRow`, the data coming from the client is parsed using `sqlbase.ParseDatumStringAsWithRawBytes`.
- it's provided `c.parsingEvalContext` as 3rd argument. this is supposed to provide timezone parameters
- however, the tz parameter in there is not configured, because:
- the pointer `c.parsingEvalContext` comes from the planner in `copyMachine`,
- which is initialized using a `resetPlanner` function provided by `(ex *connExecutor) execCopyIn`,
- ... **which does not honor the current session data.**

@jordanlewis can you triage this? Whoever gets this probably will need some guidance from @andreimatei .

**Comment by knz** [comment]
this still seems current

**Comment by rafiss** [comment]
I added a test and it seems like COPY handles the time zone as desired: https://github.com/cockroachdb/cockroach/issues/46250

**Comment by knz** [comment]
Thanks! This means that the planner plumbing has been fixed some time in between.
